app/grandchallenge/participants/forms.py

```diff
@@ -1,8 +1,110 @@
 from grandchallenge.core.widgets import JSONEditorWidget
-from grandchallenge.participants.models import RegistrationQuestion
+from grandchallenge.participants.form_fields import (
+    RegistrationQuestionAnswerField,
+)
+from grandchallenge.participants.models import (
+    RegistrationQuestion,
+    RegistrationQuestionAnswer,
+    RegistrationRequest,
+)
+
+
+class RegistrationRequestForm(ModelForm):
+
+    class Meta:
+        model = RegistrationRequest
+        fields = (
+            "user",
+            "challenge",
+        )
+        widgets = {
+            "user": HiddenInput(),
+            "challenge": HiddenInput(),
+        }
+
+    def __init__(self, *args, challenge, user, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields["user"].initial = user
+        self.fields["user"].disabled = True
+        self.fields["challenge"].initial = challenge
+        self.fields["challenge"].disabled = True
+
+        self.answer_formset = self._get_answer_formset(
+            challenge,
+            data=kwargs.get("data"),
+        )
+
+    def _get_answer_formset(self, challenge, data):
+        questions = challenge.registration_questions.all()
+        answer_formset_factory = inlineformset_factory(
+            parent_model=self._meta.model,
+            model=RegistrationQuestionAnswer,
+            form=RegistrationQuestionAnswerForm,
+            max_num=questions.count(),
+            min_num=questions.count(),
+            absolute_max=questions.count(),
+            extra=0,
+            can_delete=False,
+            can_delete_extra=False,
+        )
+        return answer_formset_factory(
+            instance=self.instance,
+            data=data,
+            initial=[{"question": q} for q in questions],
+        )
+
+    def is_valid(self):
+        return super().is_valid() and self.answer_formset.is_valid()
+
+    def full_clean(self, *args, **kwargs):
+        super().full_clean(*args, **kwargs)
+        self.answer_formset.full_clean()
+
+        # Show formset-level validation errors
+        for error in self.answer_formset.non_form_errors():
+            self.add_error(field=None, error=error)
+
+    def save(self, *args, **kwargs):
+        registration_request = super().save(*args, **kwargs)
+
+        # By default, formsets only save changed or non-empty
+        # forms. Short-circuit saving to ensure it saves empty
+        # answers
+        for form in self.answer_formset.forms:
+            self.answer_formset.save_new(form)
+
+        return registration_request
+
+
+class RegistrationQuestionAnswerForm(ModelForm):
+
+    class Meta:
+        model = RegistrationQuestionAnswer
+        fields = ("answer",)
+
+    def __init__(self, *args, initial, **kwargs):
+
+        # Formsets disabled the use of the attribute by default,
+        # since we don't allow for 'extra' forms
+        # we can forcefully allow it.
+        kwargs["use_required_attribute"] = True
```

Comment:
I think this then assumes that this form will only be used in a formset. Can you pass it as a kwarg only where necessary? See `form_kwargs` in:

```python
self.options_form_set = options_form_set_factory(
            instance=self.instance,
            data=kwargs.get("data"),
            form_kwargs={"is_editable": self.instance.is_fully_editable},
        )
```

Reply:
Good suggestion! I ignored the form_kwargs when I saw this:

https://github.com/django/django/blob/1857b6663b664847a426e8c79b8aec42994660ba/django/forms/formsets.py#L259-L270

But then I realized it was actually done here, and allows the kwarg forms to overwrite.

https://github.com/django/django/blob/1857b6663b664847a426e8c79b8aec42994660ba/django/forms/formsets.py#L219-L246

Caution for the future: the `form_kwargs` route only works when you do not allow new empty forms.